### PR TITLE
AMQP connector

### PR DESCRIPTION
**Pull request by johanandren**
Refs #41

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,3 +6,7 @@ lazy val contrib = project.
 
 lazy val mqtt = project.
   enablePlugins(AutomateHeaderPlugin)
+
+lazy val amqp = project.
+  enablePlugins(AutomateHeaderPlugin)
+  .configs(IntegrationTest)
```
Comment (johanandren) [inline build.sbt]: How can I add a config in `amqp/build.sbt`? Couldn't figure it out.

Comment (2m) [inline build.sbt]: Was not sure as well. After a bit of poking around, it looks like we can set it in `amqp/build.sbt` by adding:

```scala
lazy val amqp = (project in file(".")).
  configs(IntegrationTest)
  enablePlugins(AutomateHeaderPlugin)
```

Enabling plugins work there as well. So we can leave `~/build.sbt` with only

```scala
lazy val amqp = project
```

File: amqp/src/it/scala/akka/stream/contrib/amqp/README.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+All these tests require a locally running amqp server that does not require any authorization
```
Comment (patriknw) [inline amqp/src/it/scala/akka/stream/contrib/amqp/README.txt]: how do we make it run on Travis?

Comment (johanandren) [inline amqp/src/it/scala/akka/stream/contrib/amqp/README.txt]: Not sure we can, since it needs a rabbit mq installed. I'll look into it.

Comment (johanandren) [inline amqp/src/it/scala/akka/stream/contrib/amqp/README.txt]: Maybe: https://docs.travis-ci.com/user/database-setup/#RabbitMQ

File: amqp/src/main/scala/akka/stream/contrib/amqp/AmqpSink.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      })
+      pull(in)
+    }
+
+    setHandler(in, new InHandler {
+      override def onPush(): Unit = {
+        val elem = grab(in)
+        channel.basicPublish(
+          settings.exchange.getOrElse(""),
+          settings.routingKey.getOrElse(""),
```
Comment (patriknw) [inline amqp/src/main/scala/akka/stream/contrib/amqp/AmqpSink.scala]: does those change after construction? place them in a `val`?

File: amqp/src/main/scala/akka/stream/contrib/amqp/AmqpSource.scala [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+          queue.headOption.foreach(message =>
+            channel.basicNack(message.envelope.getDeliveryTag, true, true))
+        }
+      }
+
+    })
+
+    def pushAndAckMessage(message: IncomingMessage): Unit = {
+      push(out, message)
+      // ack it as soon as we have passed it downstream
```
Comment (patriknw) [inline amqp/src/main/scala/akka/stream/contrib/amqp/AmqpSource.scala]: yeah, here a future improvement would be to do something like the kafka offset commits

**Comment by patriknw** [comment]
LGTM, great work @johanandren

File: .travis.yml [hunk 5]
```diff
@@ -8,5 +8,6 @@ cache:
   directories:
     - $HOME/.ivy2/cache
     - $HOME/.sbt/boot
+sudo: required
```
Comment (ktoso) [inline .travis.yml]: yup

**Comment by johanandren** [comment]
Yay, integration tests running on travis, and passing.

**Comment by 2m** [comment]
LGTM

**Pull request merged**